main.cpp

```diff
@@ -92,7 +92,128 @@ good to go!
  */
 
 #include <iostream>
-int main() 
+
+class FloatType
+{
+public:
+    FloatType()
+    {
+    }
+    ~FloatType()
+    {
+    }
+
+    float add( float lhs, float rhs )
+    {
+           return lhs + rhs;
+    }
+
+    float subtract( float lhs, float rhs )
+    {
+        return lhs - rhs;
+    }
+
+    float multiply( float lhs,float rhs )
+    {
+        return lhs * rhs;
+    }
+
+    float divide( float lhs,float rhs )
+    {
+        if (rhs == 0.0f)
+        std::cout << "\n" << "warning, floating point division by zero returns 'inf' !" << std::endl;
+        return lhs / rhs;
+    }
+
+private:
+
+};
+
+class DoubleType
+{
+public:
+    DoubleType()
+    {
+    }
+    ~DoubleType()
+    {
+    }
+
+    double add( double lhs, double rhs )
+    {
+        return lhs + rhs;
+    }
+
+    double subtract( double lhs, double rhs )
+    {
+        return lhs - rhs;
+    }
+
+    double multiply( double lhs,double rhs )
+    {
+        return lhs * rhs;
+    }
+
+    double divide( double lhs,double rhs )
+    {
+        if (rhs == 0.0)
+            std::cout << "\n" << "warning, floating point division by zero returns 'inf' !" << std::endl;
+        
+        return lhs / rhs;
+    }
+
+private:
+
+};
+
+class IntType
+{
+public:
+    IntType()
+    {
+    }
+    ~IntType()
+    {
+    }
+
+    int add( int lhs, int rhs )
+    {
+        return lhs + rhs;
+    }
+
+    int subtract( int lhs, int rhs )
+    {
+        return lhs - rhs;
+    }
+
+    int multiply( int lhs,int rhs )
+    {
+        return lhs * rhs;
+    }
+
+    int divide( int lhs,int rhs )
+    {
+        if (rhs == 0)
+        {
+           std::cout << "error, integer division by zero will crash the program!" << std::endl;
+           std::cout << "returning lhs" << std::endl;
+           return lhs;
+        }
+        else
+        {
+            return lhs / rhs;
+        }
```

Comment:
If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
if the `if()` block is entered, the `return aValue;` is the last line of code to be executed.
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
`return` statements are the last thing that **ever happens** in a function
That means if your `if()` block is **NOT** entered, it will be skipped.  What is the next thing that happens?  `return anotherValue;`
that means you do **NOT** need the else block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    return anotherValue;
}
```
- [ ] revise